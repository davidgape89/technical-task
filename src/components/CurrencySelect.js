import React from 'react';
import propTypes from 'prop-types';
import Select from '@material-ui/core/Select';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import MenuItem from '@material-ui/core/MenuItem';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';

export const CurrencySelect = ({
  currencies,
  onChange,
  value,
}) => {
  const balance = Math.round(currencies[value] * 100) / 100;
  return (
    <FormControl fullWidth>
      <Select
        value={value}
        onChange={event => onChange(event.target.value)}
        input={
          <OutlinedInput labelWidth={0} />
        }
      >
        {Object.keys(currencies).map(currency => (
          <MenuItem
            key={currency}
            value={currency}
          >
            {currency}
          </MenuItem>
        ))}
      </Select>
      <FormHelperText>
Balance:
        {balance}
      </FormHelperText>
    </FormControl>
  );
};

CurrencySelect.propTypes = {
  currencies: propTypes.object,
  onChange: propTypes.func,
  value: propTypes.string,
};

