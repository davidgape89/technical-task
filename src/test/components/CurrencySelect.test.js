import React from 'react';
import {shallow} from 'enzyme';

import currencies from '../fixtures/pockets';
import Select from '@material-ui/core/Select';
import CurrencySelect from '../../components/CurrencySelect';

describe('CurrencySelect -', () => {
  let wrapper, onChange, defaultProps;

  beforeEach(() => {
    onChange = jest.fn();
    defaultProps = {
      currencies,
      value: 'EUR',
      onChange: onChange
    };
    wrapper = shallow(
      <CurrencySelect {...defaultProps} />
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('changes the currency when selected', () => {
    const event = {
      target: {
        value: 'USD'
      }
    };
    wrapper.find(Select).prop('onChange')(event);
    expect(onChange).toHaveBeenCalledWith('USD');
  });
});