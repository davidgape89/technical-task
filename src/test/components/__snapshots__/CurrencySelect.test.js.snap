// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelect - renders correctly 1`] = `
<WithStyles(FormControl)
  fullWidth={true}
>
  <WithStyles(WithFormControlContext(Select))
    input={
      <WithStyles(OutlinedInput)
        labelWidth={0}
      />
    }
    onChange={[Function]}
    value="EUR"
  >
    <WithStyles(MenuItem)
      key="EUR"
      value="EUR"
    >
      EUR
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="USD"
      value="USD"
    >
      USD
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="GBP"
      value="GBP"
    >
      GBP
    </WithStyles(MenuItem)>
  </WithStyles(WithFormControlContext(Select))>
  <WithStyles(WithFormControlContext(FormHelperText))>
    Balance: 
    20
  </WithStyles(WithFormControlContext(FormHelperText))>
</WithStyles(FormControl)>
`;
